src/hacbs/components/PipelineRunListView/PipelineRunsListView.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+import {
+  Bullseye,
+  Button,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  EmptyStateSecondaryActions,
+  Spinner,
+  Title,
+  Toolbar,
+  ToolbarContent,
+  ToolbarItem,
+} from '@patternfly/react-core';
+import { OutlinedFileImageIcon } from '@patternfly/react-icons/dist/esm/icons/outlined-file-image-icon';
+import { NamespaceContext } from '../../../components/NamespacedPage/NamespacedPage';
+import { Table } from '../../../shared';
+import { PipelineRunGroupVersionKind } from '../../models';
+import { PipelineRunKind } from '../../types';
+import { PipelineRunListHeader } from './PipelineRunListHeader';
+import PipelineRunListRow from './PipelineRunListRow';
+
+type PipelineRunsListViewProps = { applicationName: string };
+const PipelineRunsListView: React.FC<PipelineRunsListViewProps> = ({ applicationName }) => {
+  const { namespace } = React.useContext(NamespaceContext);
+
+  const [pipelineRuns, loaded] = useK8sWatchResource<PipelineRunKind[]>({
+    groupVersionKind: PipelineRunGroupVersionKind,
+    namespace,
+    isList: true,
+  });
+
+  const filteredPipelineRuns = React.useMemo(
+    () =>
+      pipelineRuns?.filter(
+        (p) => p.metadata.labels['build.appstudio.openshift.io/application'] === applicationName,
+      ),
+    [pipelineRuns, applicationName],
+  );
+  filteredPipelineRuns?.sort(
```

Comment:
nit: May as well move this down to the conditional block where we know we have pipelineruns loaded.